server/controllers/finance/reports/clientsReport/index.js

```diff
@@ -0,0 +1,201 @@
+/**
+ * Clients report Controller
+ *
+ *
+ * This controller is responsible for processing Clients report.
+ *
+ * @module finance/clientsReport
+ *
+ * @requires lodash
+ * @requires lib/db
+ * @requires lib/ReportManager
+ */
+
+'use strict';
+
+const _          = require('lodash');
+
+const db         = require('../../../../lib/db');
+const ReportManager = require('../../../../lib/ReportManager');
+
+const TEMPLATE = './server/controllers/finance/reports/clientsReport/report.handlebars';
+
+/**
+ * @function document
+ * @description processes and renders the client report document
+ */
+function document(req, res, next) {
+  const session = {};
+  const params = req.query;
+  let report;
+
+  session.dateFrom = new Date(params.dateFrom);
+  session.dateTo = new Date(params.dateTo);
+  session.detailPrevious = params.detailPrevious;
+  session.ignoredClients = params.ignoredClients;
+  session.enterprise = req.session.enterprise;
+
+  _.defaults(params, {user : req.session.user });
+
+  try {
+    report = new ReportManager(TEMPLATE, req.session, params);
+  } catch (e) {
+    return next(e);
+  }
+
+  //Getting data to be rendered
+  fetchClientsData(session)
+    .then(function (data) {
+      return report.render(data);
+    })
+    .then(function (result) {
+      res.set(result.headers).send(result.report);
+    })
+    .catch(next);
+ }
+
+/**
+ * @function fetchClientsData
+ * @description
+ * Fetch client data for current and previous fiscal year
+ **/
+function fetchClientsData(session) {  
+  //will contain data to fill in the report
+  var clientsData = {};
+  var ignoredClients, ignoredToken = '';
+
+  //attaching the enterprise to the context (good ???)
+  clientsData.enterprise = session.enterprise;
+
+  //Attaching the describing choice to the context (true | false)
+  clientsData.detailPrevious = session.detailPrevious;
+  
+  //Attache the dateTo and dateFrom (just attache the session??)
+  clientsData.dateFrom = session.dateFrom;
+  clientsData.dateTo = session.dateTo;
+
+  if(session.ignoredClients){
+    ignoredClients = (Array.isArray(session.ignoredClients))? session.ignoredClients : [session.ignoredClients];
+    ignoredToken = ` AND dg.uuid NOT IN (${escapeItems(ignoredClients).join(',')})`;
```

Comment:
Angular sends an array if there are many client to ignore and a simple string if there is only one client to ignore, that why I used that.